diff -r eb9bf73f8f7d Framework/BGHUDButtonCell.m
--- a/Framework/BGHUDButtonCell.m	Sun Jun 27 22:08:11 2010 +0300
+++ b/Framework/BGHUDButtonCell.m	Mon Jun 28 18:58:41 2010 +0300
@@ -185,6 +185,16 @@
 							 range: NSMakeRange(0, [newTitle length])];
 		}
 		
+		//--Modified 2009-11-17 by Alun Bestor - use the shadow from the current theme
+		NSShadow *textShadow = [[[BGThemeManager keyedManager] themeForKey: self.themeKey] textShadow];
+		if (textShadow)
+		{
+			[newTitle addAttribute: NSShadowAttributeName
+							 value: textShadow
+							 range: NSMakeRange(0, [newTitle length])];
+		}
+		//--End of modifications
+		
 		[newTitle endEditing];
 		
 		//Make the super class do the drawing
diff -r eb9bf73f8f7d Framework/BGHUDLabel.m
--- a/Framework/BGHUDLabel.m	Sun Jun 27 22:08:11 2010 +0300
+++ b/Framework/BGHUDLabel.m	Mon Jun 28 18:58:41 2010 +0300
@@ -93,6 +93,23 @@
 		[self setTextColor: [[[BGThemeManager keyedManager] themeForKey: self.themeKey] disabledTextColor]];
 	}
 	
+	//--Added 2009-11-17 by Alun Bestor - add text shadow before we draw the text itself
+	NSShadow *textShadow = [[[BGThemeManager keyedManager] themeForKey: self.themeKey] textShadow];
+	if (textShadow)
+	{
+		NSMutableAttributedString *value;
+		if ([self attributedStringValue])	value = [[self attributedStringValue] mutableCopy];
+		else								value = [[NSMutableAttributedString alloc] initWithString: [self stringValue]];
+	
+		[value addAttribute: NSShadowAttributeName
+					  value: textShadow
+					  range: NSMakeRange(0, [value length])];
+	
+		[self setAttributedStringValue: value];
+		[value release];
+	}
+	//--End of modifications
+	
 	[super drawRect: rect];
 }
 
diff -r eb9bf73f8f7d Framework/BGHUDSegmentedCell.m
--- a/Framework/BGHUDSegmentedCell.m	Sun Jun 27 22:08:11 2010 +0300
+++ b/Framework/BGHUDSegmentedCell.m	Mon Jun 28 18:58:41 2010 +0300
@@ -241,7 +241,19 @@
 	NSMutableDictionary *textAttributes = [[NSMutableDictionary alloc] initWithCapacity: 0];
 	
 	[textAttributes setValue: [NSFont controlContentFontOfSize: [NSFont systemFontSizeForControlSize: [self controlSize]]] forKey: NSFontAttributeName];
-	[textAttributes setValue: [[[BGThemeManager keyedManager] themeForKey: self.themeKey] textColor] forKey: NSForegroundColorAttributeName];
+	
+	//--Added 2010-06-28 by Alun Bestor to show disabled segments as properly disabled
+	NSColor *textColor;
+	if ([self isEnabledForSegment: segment])
+	{
+		textColor = [[[BGThemeManager keyedManager] themeForKey: self.themeKey] textColor];
+	}
+	else
+	{
+		textColor = [[[BGThemeManager keyedManager] themeForKey: self.themeKey] disabledTextColor];
+	}
+	[textAttributes setValue: textColor forKey: NSForegroundColorAttributeName];
+	//--End of modifications
 	
 	if([self labelForSegment: segment]) {
 		
diff -r eb9bf73f8f7d Framework/BGTheme.h
--- a/Framework/BGTheme.h	Sun Jun 27 22:08:11 2010 +0300
+++ b/Framework/BGTheme.h	Mon Jun 28 18:58:41 2010 +0300
@@ -102,6 +102,11 @@
 -(NSColor *)textColor;							//Color used to draw text
 -(NSColor *)disabledTextColor;					//Color used to draw disabled text
 -(NSShadow *)dropShadow;						//Drop shadow used on all controls
+
+//--Added 2009-11-17 by Alun Bestor to add drop shadows for text also
+-(NSShadow *)textShadow;
+//--End of modifications
+
 -(CGFloat)alphaValue;								//Alpha value that is an optional paremeter, used only
 												//if you want certain colors to be draw with the same
 												//alpha value.
diff -r eb9bf73f8f7d Framework/BGTheme.m
--- a/Framework/BGTheme.m	Sun Jun 27 22:08:11 2010 +0300
+++ b/Framework/BGTheme.m	Mon Jun 28 18:58:41 2010 +0300
@@ -344,6 +344,10 @@
 	return [shadow autorelease];
 }
 
+//--Added 2009-11-17 by Alun Bestor - override this in your own themes to render shadows on NSTextFields and NSButtons.
+-(NSShadow *)textShadow { return nil; }
+//--End of modifications
+
 -(CGFloat)alphaValue {
 	
 	return 0.6f;
